how-to-update-a-multi-regs-notice-with-translations/src/commons.js

```diff
@@ -18,25 +18,33 @@ const fetchAPIToken = async () => {
     return response.data.access_token;
   } catch (error) {
     console.error(error);
+    throw error;
   }
 };
 
 /**
- * Return a translatable text with only the default (EN) language.
+ * Returns a language-specific value from the given object.
+ *
+ * If the environment variable `SPECIFIC_LANGUAGE` is set to "true", it will first
+ * attempt to retrieve `object[language]`. If not found, or if `SPECIFIC_LANGUAGE` is not "true",
+ * it returns `object.en` as the fallback.
+ *
+ * The language specific option is used when replacing macros.
+ *
  */
-const getDefaultLanguage = (object) => object?.en;
+const getDefaultLanguage = (object, language) =>
```

Comment:
It's there an instance where you pass a language and `SPECIFIC_LANGUAGE` is false? Otherwise you can simply use `language` for the condition and default to English.

Also, I think adding a language as a parameter renders the name of the function misleading. Because you're not only getting the default language anymore. I know it's a big change but I'd consider renaming the function to something like `getLanguageOrDefault()`. This is up to you.